Refactor molecules

Luthaf opened the pull request:
Simple refactoring of the molecule type and associated structs.

g-bauer commented on src/core/src/sys/connect.rs:
```diff
@@ -0,0 +1,163 @@
+//! Data about bonds and angles in the system.
+use std::cmp::{min, max};
+
+/// A `Bond` between the atoms at indexes `i` and `j`
```
I'd go with `particles` instead of `atoms` in all descriptions. Makes it easier to understand since we also have the `Particle` struct.

g-bauer commented on src/core/src/sys/molecules.rs:
```diff
@@ -2,141 +2,15 @@
 // Copyright (C) 2015-2016 G. Fraux — BSD license
 
 //! Data about bonds and angles in the system.
```
Data about molecules in the system

g-bauer commented on src/core/src/sys/molecules.rs:
```diff
@@ -154,10 +28,8 @@ pub struct Molecule {
-    /// Index of the last (included) atom in this molecule.
-    last: usize,
+    /// Range of atomic indexes in this molecule.
+    atoms: Range<usize>,
```
Maybe rename this to something like `particles_range`? I think `particles` would also work but could lead to confusion.

g-bauer commented on src/core/src/sys/molecules.rs:
```diff
@@ -176,39 +48,39 @@ impl Molecule {
-    pub fn last(&self) -> usize {
-        self.last
+    /// Get the index of the first atom after this molecule
+    pub fn next_atom(&self) -> usize {
```
I think the name is misleading. This reads somewhat like an iterator, returning the next atom of  `self`.

Luthaf replied:
How about `end` then, like the std `Range`? The point is that this value is the first index not in the molecule.

Luthaf commented:
Rebased and corrected.

The codecov status is weird, it looks like it is now considering closing braces as (sometimes not coverged) lines of there own, while it was not before. [before](https://codecov.io/gh/lumol-org/lumol/src/1645f56aa944ef4afbe212354fa0a45e31532ee3/src/core/src/out.rs#L125) - [after](https://codecov.io/gh/lumol-org/lumol/src/92808b4bb37c7cdb552d25ae2c3f4be88d46bda6/src/core/src/out.rs#L125). What should we do? I can try to add tests to bring coverage up, or we can relax the target value and set it to 84%.

g-bauer commented:
Well, I think - for now - we could just lower the coverage target. I think it would be good to specifically
 look at adding (good) tests before we go public. I'd like to merge, if you're ok with it. 👍

Luthaf commented:
> I'd like to merge, if you're ok with it. 👍

Just let me change coverage target before, and we'll be good.
